ctpolicy/chromepolicy_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package ctpolicy
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+func wantedGroups(goog int, nonGoog int, base int) map[string]*LogGroupInfo {
+	gi := map[string]*LogGroupInfo{
+		"Google-operated": {
+			name: "Google-operated",
+			LogURLs: map[string]bool{
+				"ct.googleapis.com/aviator/":   true,
+				"ct.googleapis.com/icarus/":    true,
+				"ct.googleapis.com/rocketeer/": true,
+				"ct.googleapis.com/racketeer/": true,
+			},
+			minInclusions: goog,
+			isBase:        false,
+		},
+		"Non-Google-operated": {
+			name: "Non-Google-operated",
+			LogURLs: map[string]bool{
+				"log.bob.io": true,
+			},
+			minInclusions: nonGoog,
+			isBase:        false,
+		},
+		"All-logs": {
+			name: "All-logs",
+			LogURLs: map[string]bool{
+				"ct.googleapis.com/aviator/":   true,
+				"ct.googleapis.com/icarus/":    true,
+				"ct.googleapis.com/rocketeer/": true,
+				"ct.googleapis.com/racketeer/": true,
+				"log.bob.io":                   true,
+			},
+			minInclusions: base,
+			isBase:        true,
+		},
+	}
+	return gi
+}
+
+func TestCheckChromePolicy(t *testing.T) {
+	tests := []struct {
+		name string
+		cert *x509.Certificate
+		want map[string]*LogGroupInfo
+	}{
+		{
+			name: "Short",
+			cert: getTestCertPEMShort(),
+			want: wantedGroups(1, 1, 2),
+		},
+		{
+			name: "2-year",
+			cert: getTestCertPEM2Years(),
+			want: wantedGroups(1, 1, 3),
+		},
+		{
+			name: "3-year",
+			cert: getTestCertPEM3Years(),
+			want: wantedGroups(1, 1, 4),
+		},
+		{
+			name: "Long",
+			cert: getTestCertPEMLongOriginal(),
+			want: wantedGroups(1, 1, 5),
+		},
+	}
+
+	var policy ChromeCTPolicy
+	sampleLogList := sampleLogList(t)
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, err := policy.LogsByGroup(test.cert, sampleLogList)
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("LogsByGroup returned %v, want %v", got, test.want)
+			}
+			if err != nil {
```

Comment:
Needs a test for the case when a non-`nil` error and a set of groups are both returned.

Reply:
Test added.